### PR TITLE
main_51_sonarLint-Fix BackEnd

### LINKED ISSUE #51: Fix Sonar issues and formatting problems in the code base

### PR DESCRIPTION
**Pull request by janithkw**
(Needs review)

### CODE REVIEW

File: ocrspring/src/main/java/com/oasis/ocrspring/controller/Controller.java [hunk 1]
```diff
@@ -52,28 +52,30 @@ public class Controller {
...
-//    @GetMapping("/allPatientDetails")
-//    public List<Patient> AllPatientDetails(){
+
+    //    @GetMapping("/allPatientDetails")
+//    public List<Patient> allPatientDetails(){
 //
 //        System.out.println("Get all user details");
-//        return patientService.AllPatientDetails();
+//        return patientService.allPatientDetails();
 //    }
```
Comment (AchinthaHarshamal) [inline ocrspring/src/main/java/com/oasis/ocrspring/controller/Controller.java]: remove this comments

File: ocrspring/src/main/java/com/oasis/ocrspring/controller/Controller.java [hunk 2]
```diff
@@ -85,64 +87,70 @@ public List<Request> AllRequestDetails() {
...
+        return optionService.allOptionDetails();
     }
 
-//    @GetMapping("/allImageDetails")
-//    public List<Image> AllImageDetails() {
+    //    @GetMapping("/allImageDetails")
+//    public List<Image> allImageDetails() {
 //
-//        return imageService.AllImageDetails();
+//        return imageService.allImageDetails();
```
Comment (AchinthaHarshamal) [inline ocrspring/src/main/java/com/oasis/ocrspring/controller/Controller.java]: remove unnecessary comments

File: ocrspring/src/main/java/com/oasis/ocrspring/controller/Controller.java [hunk 3]
```diff
@@ -85,64 +87,70 @@ public List<Request> AllRequestDetails() {
...
-        return draftReportService.AllDraftReportDetails();
+        return draftReportService.allDraftReportDetails();
     }
+
     @PostMapping("/postUserDetails")
-    public User postUserDetails(@RequestBody User user){
+    public User postUserDetails(@RequestBody User user) {
         return userservice.createUser(user);
     }
 //    @PostMapping("/signup")
```
Comment (AchinthaHarshamal) [inline ocrspring/src/main/java/com/oasis/ocrspring/controller/Controller.java]: remove

File: ocrspring/src/main/java/com/oasis/ocrspring/controller/UserController.java [hunk 4]
```diff
@@ -48,53 +47,47 @@ public void setMessage(String message) {
...
     }
+
     @PostMapping("/update")
-    public ResponseEntity<?> updateUserDetail(String _id, @RequestBody UserDto userBody){////////////////////////////need to get id part through authentication
-        try{
-            Optional<User> existingUser = userservice.getUserById(_id);
-            if(existingUser.isPresent()){
-                //Optional<User> updatedUser = userservice.getUserById(_id);
-                User updatedUser= userservice.updateUser(_id,userBody);
+    public ResponseEntity<?> updateUserDetail(String id, @RequestBody UserDto userBody) {////////////////////////////need to get id part through authentication
```
Comment (AchinthaHarshamal) [inline ocrspring/src/main/java/com/oasis/ocrspring/controller/UserController.java]: add as TODO:

File: ocrspring/src/main/java/com/oasis/ocrspring/dto/ImageRequestDto.java [hunk 5]
```diff
@@ -1,73 +1,23 @@
...
 public class ImageRequestDto {
     @JsonProperty("telecon_entry_id")
     private String teleconId;
     @JsonProperty("image_name")
-    private String image_name;
+    private String imageName;
     private String location;
-    private String clinical_diagnosis;
-    private Boolean lesions_appear;
+    private String clinicalDiagnosis;
```
Comment (AchinthaHarshamal) [inline ocrspring/src/main/java/com/oasis/ocrspring/dto/ImageRequestDto.java:17]: check this again

File: ocrspring/src/main/java/com/oasis/ocrspring/dto/ImageRequestDto.java [hunk 6]
```diff
@@ -1,73 +1,23 @@
...
+    private Boolean lesionsAppear;
```
Comment (AchinthaHarshamal) [inline ocrspring/src/main/java/com/oasis/ocrspring/dto/ImageRequestDto.java:18]: check this again

File: ocrspring/src/main/java/com/oasis/ocrspring/dto/PatientDetailsDto.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
+package com.oasis.ocrspring.dto;
```
Comment (AchinthaHarshamal) [inline ocrspring/src/main/java/com/oasis/ocrspring/dto/PatientDetailsDto.java:1]: why there is a new class..

File: ocrspring/src/main/java/com/oasis/ocrspring/dto/PatientTeleconResponse.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
+package com.oasis.ocrspring.dto;
```
Comment (AchinthaHarshamal) [inline ocrspring/src/main/java/com/oasis/ocrspring/dto/PatientTeleconResponse.java:1]: why there is a new class

File: ocrspring/src/main/java/com/oasis/ocrspring/dto/ReportsRequestDto.java [hunk 9]
```diff
@@ -1,26 +1,16 @@
...
 import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
 
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
 public class ReportsRequestDto {
     @JsonProperty("telecon_entry_id")
```
Comment (AchinthaHarshamal) [inline ocrspring/src/main/java/com/oasis/ocrspring/dto/ReportsRequestDto.java]: check need of JsonProperty

File: ocrspring/src/main/java/com/oasis/ocrspring/repository/RequestRepository.java [hunk 10]
```diff
@@ -1,13 +1,13 @@
...
 import org.springframework.data.mongodb.repository.MongoRepository;
 
 import java.util.Optional;
 
-public interface RequestRepository extends MongoRepository<Request, String>{
+public interface RequestRepository extends MongoRepository<Request, String> {
     Optional<Request> findByRegNo(String reg_no);
+
     Optional<Request> findByEmail(String email);
 //    Optional<Request> findByRegNo(String regNo);
```
Comment (AchinthaHarshamal) [inline ocrspring/src/main/java/com/oasis/ocrspring/repository/RequestRepository.java]: remove this

File: ocrspring/src/main/java/com/oasis/ocrspring/service/ImageService.java [hunk 11]
```diff
@@ -42,8 +42,8 @@ public ResponseEntity<UploadImageResponse> uploadImages(ImageRequestDto data,
                                                             String id,
                                                             List<MultipartFile> files) throws IOException {
         List<Image> uploadedImages = new ArrayList<>();
-        List<String> ImageIds = new ArrayList<>();
-        List<String> ImageURIs = new ArrayList<>();
+        List<String> imageIds_ = new ArrayList<>();
```
Comment (AchinthaHarshamal) [inline ocrspring/src/main/java/com/oasis/ocrspring/service/ImageService.java]: why there is _

File: ocrspring/src/main/java/com/oasis/ocrspring/service/ResponseMessages/ErrorResponse.java [hunk 12]
```diff
@@ -8,6 +8,7 @@ public ErrorResponse(boolean success, String message) {
         this.success = success;
         this.message = message;
     }
+
     // Getters and setters
```
Comment (AchinthaHarshamal) [inline ocrspring/src/main/java/com/oasis/ocrspring/service/ResponseMessages/ErrorResponse.java]: irrelevant comment

File: ocrspring/src/main/java/com/oasis/ocrspring/service/TeleconEntriesService.java [hunk 13]
```diff
@@ -25,37 +25,40 @@
...
 
-        try{
-            Patient patient = patientService.findPatient(patient_id,clinician_id);//patient_id, String clinician_id
-            if(patient != null){
+    public ResponseEntity<?> patientTeleconEntry(String patientId,
+                                                 String clinicianId,
+                                                 PatientTeleconRequest newPatient) { //path patient id
+
+        try {
+            Patient patient = patientService.findPatient(patientId, clinicianId);//patient_id, String clinician_id
```
Comment (AchinthaHarshamal) [inline ocrspring/src/main/java/com/oasis/ocrspring/service/TeleconEntriesService.java]: remove this

**Review by AchinthaHarshamal — CHANGES_REQUESTED** [review]

File: ocrspring/src/main/java/com/oasis/ocrspring/dto/PatientDetailsDto.java (hunk 7 above)
Comment (janithkw) [reply ocrspring/src/main/java/com/oasis/ocrspring/dto/PatientDetailsDto.java:1]: This is later to be used according to others

File: ocrspring/src/main/java/com/oasis/ocrspring/dto/PatientTeleconResponse.java (hunk 8 above)
Comment (janithkw) [reply ocrspring/src/main/java/com/oasis/ocrspring/dto/PatientTeleconResponse.java:1]: Later to be used

**Pull request merged**